src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -148,50 +160,92 @@ public void setWorker(final CitizenData citizen)
      * @return {@link net.minecraft.entity.Entity} of the worker
      */
     @Nullable
-    public EntityCitizen getWorkerEntity()
+    public List<EntityCitizen> getWorkerEntities()
     {
-        if (worker == null)
+        final List<EntityCitizen> entities = new ArrayList<>();
+        for (final CitizenData data : workers)
         {
-            return null;
+            if (data != null)
+            {
+                entities.add(data.getCitizenEntity());
+            }
         }
-        return worker.getCitizenEntity();
+
+        return entities;
     }
 
     @Override
     public void readFromNBT(@NotNull final NBTTagCompound compound)
     {
         super.readFromNBT(compound);
-
+        workers.clear();
         if (compound.hasKey(TAG_WORKER))
         {
-            // Bypass setWorker, which marks dirty
-            worker = getColony().getCitizen(compound.getInteger(TAG_WORKER));
-            if (worker != null)
+            try
+            {
+                final NBTTagList workersTagList = compound.getTagList(TAG_WORKER, Constants.NBT.TAG_COMPOUND);
+                for (int i = 0; i < workersTagList.tagCount(); ++i)
+                {
+                    final CitizenData data = getColony().getCitizen(workersTagList.getCompoundTagAt(i).getInteger(TAG_ID));
+                    if (data != null)
+                    {
+                        data.setWorkBuilding(this);
+                        workers.add(data);
+                    }
+                }
+            }
+            catch(final Exception e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)